File: 50 Coding Challenge Part II/TypeScript/Challenge34.ts
// Challenge 34: Create a function that will return an array with words inside a text

// Checks for the presence of a seperator
function isSeprator(char: string){
    let seperators = [" ", "\t", "\n", "\r", ",", ";", ".", "!", "?"];
    return seperators.includes(char);
}

function getWords(text: string): string[] {
    let startWord: number = -1;
    let ar: string[] = [];

    for(let i = 0; i <= text.length; i++){
        let c: string = i < text.length ? text[i] : " ";

        if (!isSeprator(c) && startWord < 0) startWord = i;

        if (isSeprator(c) && startWord >= 0){
            let word: string = text.substring(startWord, i);
            ar.push(word);

            startWord = -1;
        }
    }

    return ar;
}

File: 50 Coding Challenge Part II/Python/Challenge34.py
# Challenge 34: Create a function that will
# return an array with words inside a text
def getWords(text):
    startWords = text.split()
    words = []
    for i in startWords:
        word = []
        for j in i:
            if not isSeperator(j):
                word.append(j)
        words.append("".join(word))
    return words


def isSeperator(char):
    seperators = [" ", ",", "\t", "\n", "\r", ";", ":", ".", "!", "?", "(", ")"]
    return char in seperators


# driver Method
print(getWords(" Create a function, that will return an array (of strings), with words inside a text"))
